### PR TITLE
[O2B-1186] Add QC flag type creation page

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- Added QC Flag Type creation page

Notable changes for developers:
- NA

Changes made to the database:
- NA

### CODE REVIEW

**Comment by xsalonx** [comment]
To be merged afer https://github.com/AliceO2Group/Bookkeeping/pull/1443

File: lib/public/views/QcFlagTypes/Overview/QcFlagTypesOverviewPage.js [hunk 1]
```diff
@@ -41,6 +43,12 @@ export const QcFlagTypesOverviewPage = ({ qcFlagTypes: { overviewModel } }) => {
                 filtersPanelPopover(overviewModel, qcFlagTypesActiveColumns),
                 h('h2', 'QC Flag Types'),
             ]),
+            model.isAdmin() && [
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Overview/QcFlagTypesOverviewPage.js]: applied

[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Overview/QcFlagTypesOverviewPage.js]: applied

File: lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationPage.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+import { h, sessionService } from '/js/src/index.js';
+import { BkpRoles } from '../../../domain/enums/BkpRoles.js';
+import { creationFormComponent } from '../../../components/common/form/creationFormComponent.js';
+import { colorInputComponent } from '../../../components/common/form/inputs/colorInputComponent.js';
+import { PanelComponent } from '../../../components/common/panel/PanelComponent.js';
+import { LabelPanelHeaderComponent } from '../../../components/common/panel/LabelPanelHeaderComponent.js';
+
+/**
+ * A function to construct the create QC Flag Creation screen
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationPage.js]: applied

File: lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    /**
+     * Returns the currently inserted name from the QC Flag Type creation screen.
+     *
+     * @returns {string} The name currently inserted in the QC Flag Type creation screen.
+     */
+    get name() {
+        return this._name;
+    }
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js]: refactored

File: lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    constructor() {
+        super();
+        this._name = '';
+        this._method = '';
+        this._color = null;
+        this._bad = true;
+    }
+
+    /**
+     * Returns the current state as a plain old javascript object
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js]: refactored

File: lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    /**
+     * @inheritDoc
+     */
+    isValid() {
+        return this._data.name.length && this._data.method.length;
+    }
+}
+
+/**
+ * Data storing the current value of a QC Flag Type creation form
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js]: use `formData`

File: lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    constructor(onCreationSuccess) {
+        super('/api/qcflagTypes', ({ id }) => onCreationSuccess(id));
+    }
+
+    /**
+     * Return current data stored by model
+     */
+    get data() {
+        return this._data;
+    }
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlagTypes/Create/QcFlagTypeCreationModel.js]: use `formData`

**Review by martinboulais — APPROVED** [review]

**Pull request merged**